src/lib.rs

```diff
@@ -21,6 +21,11 @@ pub mod errors {
         pub code: ::reqwest::StatusCode
     }
 
+    #[derive(Fail, Debug)]
+    #[fail(display = "the following downloads failed:\n{}", _0)]
+    /// Files which are failed to download
```

Comment:
should we "which have" instead of "which are"

Reply:
Also, I think `DownloadsFailed` would be a better name (it's fairly clear we're downloading files!)

Reply:
> We probably want to continue handling the error in main so that it's easy to expand what Tempget does in the future.

That makes sense to me, I just propose the return type `!` for `main` which implies that `main` does not return (to anything which calls it).

Reply:
>  I think `DownloadsFailed` would be a better name (it's fairly clear we're downloading files!)

Agree!!! Changed in the updated PR.